README.md

```diff
@@ -5,40 +5,184 @@ Please edit this README.md file with your answers to the worksheet questions.
+|-------|-------|-------|---------------------------|
+| FALSE | FALSE | FALSE | FALSE                     |
+| FALSE | FALSE | TRUE  | FALSE                     |
+| FALSE | TRUE  | FALSE | FALSE                     |
+| FALSE | TRUE  | TRUE  | TRUE                      |
+| TRUE  | FALSE | FALSE | FALSE                     |
+| TRUE  | FALSE | TRUE  | TRUE                      |
+| TRUE  | TRUE  | FALSE | FALSE                     |
+| TRUE  | TRUE  | TRUE  | TRUE                      |
+
+| A     | B     | C     | A OR \(B AND C\) |
+|-------|-------|-------|------------------|
+| FALSE | FALSE | FALSE | FALSE            |
+| FALSE | FALSE | TRUE  | FALSE            |
+| FALSE | TRUE  | FALSE | FALSE            |
+| FALSE | TRUE  | TRUE  | TRUE             |
+| TRUE  | FALSE | FALSE | TRUE             |
+| TRUE  | FALSE | TRUE  | TRUE             |
+| TRUE  | TRUE  | FALSE | TRUE             |
+| TRUE  | TRUE  | TRUE  | TRUE             |
 ## Question 4
 
 ### a
+|file_exists("a.txt")|file_exists("b.txt")|First Program|Second Program|
+|--------------------|--------------------|-------------|--------------|
+|FALSE               |FALSE               |TRUE         |TRUE          |
+|FALSE               |TRUE                |TRUE         |TRUE          |
+|TRUE                |FALSE               |TRUE         |TRUE          |
+|TRUE                |TRUE                |FALSE        |FALSE         |
 
 ### b
+|x > 7|type ( x) == int|type (x) == float|First Program|Second Program|
+|-----|----------------|-----------------|-------------|--------------|
+|FALSE|FALSE           |FALSE            |FALSE        |FALSE         |
+|FALSE|FALSE           |TRUE             |FALSE        |FALSE         |
+|FALSE|TRUE            |FALSE            |FALSE        |FALSE         |
+|FALSE|TRUE            |TRUE             |TRUE         |TRUE          |
+|TRUE |FALSE           |FALSE            |FALSE        |FALSE         |
+|TRUE |FALSE           |TRUE             |FALSE        |FALSE         |
+|TRUE |TRUE            |FALSE            |TRUE         |TRUE          |
+|TRUE |TRUE            |TRUE             |TRUE         |TRUE          |
 
 ### c
+|Y==0 |X==0 |First Program|Second Program|
+|-----|-----|-------------|--------------|
+|FALSE|FALSE|FALSE        |TRUE          |
+|FALSE|TRUE |FALSE        |TRUE          |
+|TRUE |FALSE|FALSE        |TRUE          |
+|TRUE |TRUE |TRUE         |FALSE         |
 
 ### d
-
+|x > 0|X > 10|y > 0|First Program|Second Program|
+|-----|------|-----|-------------|--------------|
+|FALSE|FALSE |FALSE|FALSE        |FALSE         |
+|FALSE|FALSE |TRUE |FALSE        |FALSE         |
+|FALSE|TRUE  |FALSE|TRUE         |TRUE          |
+|FALSE|TRUE  |TRUE |TRUE         |TRUE          |
+|TRUE |FALSE |FALSE|FALSE        |FALSE         |
+|TRUE |FALSE |TRUE |TRUE         |FALSE         |
+|TRUE |TRUE  |FALSE|TRUE         |TRUE          |
+|TRUE |TRUE  |TRUE |TRUE         |TRUE          |
```

Comment:
Again right idea, but needs some explanation for why the truth tables don't match exactly (i.e. that `x>10` cannot be true if `x>0` is false so this case is not possible)